### ISSUE TITLE
investigate why 1K supervised evaluation shows as high accuracy as full training set

### ISSUE THREAD

**Issue by taeil** [Berkeley-Data/hpt#71]
- Supervised s1/s2 (1024)
- Supervised s1/s2 (full) 
- Moco s1/s2 (1024)

**Comment by taeil** [comment]
rerunning supervised training/eval to make sure it was not just recording issue.

**Comment by taeil** [comment]
rerun this eval to make sure we can reproduce. So far, cannot reproduce. 

https://wandb.ai/cal-capstone/SEN12MS/runs/rqn33k8c/overview

**Comment by taeil** [comment]
> [rerunning supervised training/eval](https://wandb.ai/cal-capstone/sup_scene_cls?workspace=user-taeil) on s1, s2, s1/s2 to make sure it was not just recording issue.

reproduced the high accuracy. 

![image](https://user-images.githubusercontent.com/3128981/114279309-c74b5f00-99e8-11eb-9e8c-f757e45f4bd5.png)

**Comment by taeil** [comment]
retrying on smaller dataset (512) for training both supervised and pretraining (visionary).

**Comment by suryatechie** [comment]
We can try the following:
1) 1024 images from the original training and not from the 31k(removed overlap images)
2) use 100 epochs

**Comment by suryatechie** [comment]
I was able to reproduce the issue. Performance was 60%. 
After using the same hyperparameters as the full pre-trained model provided by SEN12MS, I saw a decrease in performance.  Step logic that we added is adding a performance improvement by ~3.5 percent. If we don't use that and use the same hyperparameters as SEN12MS, we see that there is a decrease in performance for 1024 samples. Also, as expected, **S1 < S2 < S1S2**.
![image](https://user-images.githubusercontent.com/15736461/114286198-83208480-9a11-11eb-8b7a-53f47f492d11.png)
